api/apps/api/src/modules/projects/project-cost-surface.controller.ts

```diff
@@ -0,0 +1,132 @@
+import { plainToClass } from 'class-transformer';
+import { Response } from 'express';
+import { marxanRunFiles } from '@marxan-api/modules/scenarios/scenarios.controller';
+import { JwtAuthGuard } from '@marxan-api/guards/jwt-auth.guard';
+import { ImplementsAcl } from '@marxan-api/decorators/acl.decorator';
+
+@ApiTags(projectResource.className)
+@Controller(`${apiGlobalPrefixes.v1}/projects`)
+export class ProjectCostSurfaceController {
+  constructor(
+    @Inject(forwardRef(() => ScenariosService))
+    public readonly scenarioService: ScenariosService,
+  ) {}
+
+  @ImplementsAcl()
+  @UseGuards(JwtAuthGuard)
+  @ApiConsumesShapefile({ withGeoJsonResponse: false })
+  @GeometryFileInterceptor(GeometryKind.ComplexWithProperties)
+  @ApiTags(asyncJobTag)
+  @Post(`:scenarioId/cost-surface/shapefile`)
+  async processCostSurfaceShapefile(
+    @Param('scenarioId') scenarioId: string,
+    @Req() req: RequestWithAuthenticatedUser,
+    @UploadedFile() file: Express.Multer.File,
+  ): Promise<JsonApiAsyncJobMeta> {
+    await ensureShapefileHasRequiredFiles(file);
+
+    const result = await this.scenarioService.processCostSurfaceShapefile(
+      scenarioId,
+      req.user.id,
+      file,
+    );
+
+    if (isLeft(result)) {
+      throw mapAclDomainToHttpError(result.left, {
+        scenarioId,
+        userId: req.user.id,
+        resourceType: scenarioResource.name.plural,
+      });
+    }
+    return AsyncJobDto.forScenario().asJsonApiMetadata();
+  }
+
+  @ImplementsAcl()
+  @UseGuards(JwtAuthGuard)
+  @Get(`:scenarioId/cost-surface`)
+  @ApiOkResponse({ type: CostRangeDto })
+  async getCostRange(
+    @Param('scenarioId') scenarioId: string,
+    @Req() req: RequestWithAuthenticatedUser,
+  ): Promise<CostRangeDto> {
+    const result = await this.scenarioService.getCostRange(
+      scenarioId,
+      req.user.id,
+    );
+    if (isLeft(result)) {
+      throw mapAclDomainToHttpError(result.left, {
+        scenarioId,
+        userId: req.user.id,
+        resourceType: scenarioResource.name.plural,
+      });
+    }
+    return plainToClass<CostRangeDto, CostRangeDto>(CostRangeDto, result.right);
+  }
+
+  @ImplementsAcl()
+  @UseGuards(JwtAuthGuard)
+  @ApiTags(marxanRunFiles)
+  @Header('Content-Type', 'text/csv')
+  @ApiOkResponse({
+    schema: {
+      type: 'string',
+    },
+  })
+  @ApiOperation({
+    description: `Uploaded cost surface data`,
+  })
+  @Get(`:scenarioId/marxan/dat/pu.dat`)
+  async getScenarioCostSurface(
+    @Param('scenarioId', ParseUUIDPipe) scenarioId: string,
+    @Req() req: RequestWithAuthenticatedUser,
+    @Res() res: Response,
+  ): Promise<void> {
+    const result = await this.scenarioService.getCostSurfaceCsv(
+      scenarioId,
+      req.user.id,
+      res,
+    );
+
+    if (isLeft(result)) {
+      throw mapAclDomainToHttpError(result.left, {
+        scenarioId,
+        userId: req.user.id,
+        resourceType: scenarioResource.name.plural,
+      });
+    }
+  }
```

Comment:
this controller handler had plenty of naming confusion: in practice it should stay in `ScenariosController` and should  be renamed to something like `GetPuDatFile()`; its Swagger decorators are also fully wrong (likely copypasta from somewhere else and we didn't catch this in review)

the confusion goes on through the rabbit hole: `ScenariosController.getCostSurfaceCsv()` is likewise more like `getPuDatCsv()` and `InputFilesService.readCostSurface()` again is more like `getPuDatContent()`

(also, `InputFilesService.get<whatever>DatContent()` should be async, and they should await the async functions that they call, but well... the rabbit hole must end somewhere)

Reply:
Right, rolling this back again as I considered to be a cost-surface related operation, as the task suggest to move. 
Thanks